src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -956,3 +943,98 @@ def assign_last_name_ids(self, pop: pd.DataFrame) -> pd.Series:
         )
 
         return pop["last_name_id"]
+
+    def update_reference_person_and_relationships(
+        self, population: pd.DataFrame
+    ) -> pd.DataFrame:
+        # Takes population state table and assigns reference person in households that do not have one after a time step
+        # After a new reference person has been assigned, biological children and other household members have their
+        #  relationships updated.
+        # Find standard households that do not have a reference person
+        household_ids_with_reference_person = population.loc[
+            population["relation_to_household_head"] == "Reference person", "household_id"
+        ]
+        standard_household_ids = population.loc[
+            population["housing_type"] == "Standard", "household_id"
+        ].unique()
+        household_ids_without_reference_person = set(standard_household_ids) - set(
+            household_ids_with_reference_person
+        )
+        households_to_update_idx = population.index[
+            population["household_id"].isin(household_ids_without_reference_person)
+        ]
+
+        # Find oldest member in each household and make them new reference person
+        # This is a series with household_id as the index and the new reference person as the value
+        new_reference_persons = (
+            population.loc[households_to_update_idx].groupby(["household_id"])["age"].idxmax()
+        )
+        # Preserve old relationship of new reference person before assigning them as new reference persons
+        new_reference_persons_relationship_to_old_reference_person = population.loc[
+            new_reference_persons, ["household_id", "relation_to_household_head"]
+        ]
+        population.loc[
+            new_reference_persons, "relation_to_household_head"
+        ] = "Reference person"
+
+        # Update simulants born in simulation as biological children to their mother if mother is new reference person
+        biological_children_idx = population.index[
+            (population["date_of_birth"] > self.start_time)
+            & (
+                population["household_id"].isin(
+                    new_reference_persons_relationship_to_old_reference_person["household_id"]
+                )
+            )
+            & (
+                population["guardian_1"]
```

Comment:
Couldn't guardian_2 have become the new reference person? Are we ignoring this because you can only have a guardian 2 if guardian 1 was previously the reference person, and therefore your relationship won't change?

Reply:
@rmudambi -- I discussed this with @albrja and we decided on this, which there is a PR to document more clearly: https://github.com/ihmeuw/vivarium_research/pull/1117.

You are correct that the case with guardian_2 would be very very rare -- that household structure could only be constructed through as sequence of extremely unlikely domestic migration events.

The original intent of the docs was to not use _guardian_ here (which can -- by design -- be a grandparent or another kind of relative) but to use the _tracked parent_. At that time there was an actual column in the state table tracking the biological mother that had the fertility event. It seems that in the meantime that column was superseded by guardian(s). This is a way to "back out" who the biological mother was from the `guardian_1` column.